From 30faa95daf430f20de59f7c7bb6b2725961089bc Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:39:49 -0400
Subject: [alsa] hda: fix an external mic jack problem on a HP machine

Message-id: <140972823573-402-patchset-jkysela@redhat.com>
Patchwork-id: 91098
O-Subject: [RHEL 7.1 PATCH 401/442] ALSA: hda - fix an external mic jack problem on a HP machine
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

ON the machine, two pin complex (0xb and 0xe) are both routed to
the same external right-side mic jack, this makes the jack can't work.

To fix this problem, set the 0xe to "not connected".

BugLink: https://bugs.launchpad.net/bugs/1350148
Tested-by: Franz Hsieh <franz.hsieh@canonical.com>
Cc: stable@vger.kernel.org
Signed-off-by: Hui Wang <hui.wang@canonical.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 7440850c20b69658f322119d20a94dc914127cc7)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_sigmatel.c b/sound/pci/hda/patch_sigmatel.c
index 8dc6815..ea823e1 100644
--- a/sound/pci/hda/patch_sigmatel.c
+++ b/sound/pci/hda/patch_sigmatel.c
@@ -84,6 +84,7 @@ enum {
  STAC_DELL_EQ,
  STAC_ALIENWARE_M17X,
  STAC_92HD89XX_HP_FRONT_JACK,
+ STAC_92HD89XX_HP_Z1_G2_RIGHT_MIC_JACK,
  STAC_92HD73XX_MODELS
 };
 
@@ -1810,6 +1811,11 @@ static const struct hda_pintbl stac92hd89xx_hp_front_jack_pin_configs[] = {
  {}
 };
 
+static const struct hda_pintbl stac92hd89xx_hp_z1_g2_right_mic_jack_pin_configs[] = {
+ { 0x0e, 0x400000f0 },
+ {}
+};
+
 static void stac92hd73xx_fixup_ref(struct hda_codec *codec,
        const struct hda_fixup *fix, int action)
 {
@@ -1932,6 +1938,10 @@ static const struct hda_fixup stac92hd73xx_fixups[] = {
  [STAC_92HD89XX_HP_FRONT_JACK] = {
   .type = HDA_FIXUP_PINS,
   .v.pins = stac92hd89xx_hp_front_jack_pin_configs,
+ },
+ [STAC_92HD89XX_HP_Z1_G2_RIGHT_MIC_JACK] = {
+  .type = HDA_FIXUP_PINS,
+  .v.pins = stac92hd89xx_hp_z1_g2_right_mic_jack_pin_configs,
  }
 };
 
@@ -1992,6 +2002,8 @@ static const struct snd_pci_quirk stac92hd73xx_fixup_tbl[] = {
         "Alienware M17x", STAC_ALIENWARE_M17X),
  SND_PCI_QUIRK(PCI_VENDOR_ID_DELL, 0x0490,
         "Alienware M17x R3", STAC_DELL_EQ),
+ SND_PCI_QUIRK(PCI_VENDOR_ID_HP, 0x1927,
+    "HP Z1 G2", STAC_92HD89XX_HP_Z1_G2_RIGHT_MIC_JACK),
  SND_PCI_QUIRK(PCI_VENDOR_ID_HP, 0x2b17,
     "unknown HP", STAC_92HD89XX_HP_FRONT_JACK),
  {} /* terminator */
-- 
1.7.1